daprovider/das/local_file_storage_service.go

```diff
@@ -274,6 +274,37 @@ func listDir(dir string) ([]string, error) {
 	return files, nil
 }
 
+// readDirNamesFiltered reads directory entries at p and returns only names matching the expected type.
+// If p is not a directory or does not exist (e.g., a stray file in place of a directory), it returns
+// an empty slice and nil to skip it gracefully.
+func readDirNamesFiltered(p string, wantDirs bool) ([]string, error) {
```

Comment:
This is a nice standalone function that would be easy to write some tests for, would you mind writing some tests for it?